tests/recovery/recovery_test/recovery_test.go

```diff
@@ -168,6 +172,172 @@ func TestSnapshotFixInvalidFiles(t *testing.T) {
 	require.NoError(t, err)
 }
 
+// TestConsistencyWhenKill9AfterModify will test the data consistency while it encountered kill -9 signal.
+func TestConsistencyWhenKill9AfterModify(t *testing.T) {
+	// assumption: the test is run on filesystem & not directly on object store
+	dataRepoPath := path.Join(*repoPathPrefix, dirPath, dataPath)
+
+	baseDir := t.TempDir()
+	if baseDir == "" {
+		t.FailNow()
+	}
+
+	bm, err := blobmanipulator.NewBlobManipulator(baseDir, dataRepoPath)
+	if err != nil {
+		if errors.Is(err, kopiarunner.ErrExeVariableNotSet) {
+			t.Skip("Skipping crash consistency tests because KOPIA_EXE is not set")
+		} else {
+			t.Skip("Error creating SnapshotTester:", err)
+		}
+	}
+
+	bm.DataRepoPath = dataRepoPath
+
+	// create a snapshot for initialized data
+	snapID, err := bm.SetUpSystemWithOneSnapshot()
+	if err != nil {
+		t.FailNow()
+	}
+
+	cmpDir := bm.PathToTakeSnapshot
+
+	copyDir := t.TempDir()
+	if copyDir == "" {
+		t.FailNow()
+	}
+
+	err = bm.FileHandler.CopyAllFiles(cmpDir, copyDir)
+	require.NoError(t, err)
+
+	// add files
+	fileSize := 1 * 1024 * 1024
+	numFiles := 200
+
+	err = bm.GenerateRandomFiles(fileSize, numFiles)
+	require.NoError(t, err)
+
+	err = bm.FileHandler.CopyAllFiles(bm.PathToTakeSnapshot, copyDir)
+	require.NoError(t, err)
+
+	// modify original files
+	content := "\nthis is a test for TestConsistencyWhenKill9AfterModify\n"
+	err = bm.FileHandler.ModifyDataSetWithContent(copyDir, content)
+	require.NoError(t, err)
+	log.Println("Copy content and modify the files.")
```

Comment:
You should use `t.Log()` and `t.Logf()` where possible to avoid test noise.

SEE: https://pkg.go.dev/testing#B.Logf

These logs are generally not printed on the console, unless you specify the verbose `-v` flag to the test. Like `go test -v`.

The reason for using `t.Log` and `t.Logf` over `log.PrintX` is because developers usually run all the tests quite often. Every time we push anything to Kopia, we'd run all these tests locally. And if these tests would keep printing noise to the screen then it is hard to follow progress of all the tests. If anyone is interested in watching what a test is doing then they can specify the verbose flag or if the test fails then the test framework will dump the Log() statements anyway, so that they are there when you need them.

Reply:
That's why I find it confusing when I see `t.Log` used in some places and `log.Println` used in others. I appreciate your response.